### PR TITLE
Port PT Profiler to habana_main

### PR DESCRIPTION
**Pull request by adobrzyniewicz-habana**
Porting PT Profiler from:
https://github.com/HabanaAI/vllm-fork/commit/81a23a708195faef6167919890cefa225a721907
and
https://github.com/HabanaAI/vllm-fork/commit/e805b885d32a749d9409f13b6446895d13e8b885

### CODE REVIEW

File: vllm/worker/habana_model_runner.py [hunk 1]
```diff
@@ -183,6 +184,75 @@ def align_workers(value, op):
     return value_t.item()
 
 
+def setup_profiler():
```
Comment (michalkuligowski) [inline vllm/worker/habana_model_runner.py:187]: This can be removed, since its overridden in 245

Comment (adobrzyniewicz-habana) [reply vllm/worker/habana_model_runner.py]: Removed

File: vllm/worker/habana_model_runner.py [hunk 2]
```diff
@@ -210,6 +211,56 @@ def align_workers(value, op):
...
+        def start(self):
+            self.cur_step = 0
+            self.check()
+        def step(self):
+            self.cur_step = self.cur_step + 1
+            self.check()
+        def stop(self):
+            api.profiler_stop(TraceType.TraceAll, 0)
+            api.profiler_get_trace_json(TraceType.TraceAll, 0)
+    return SynapseProfiler()
```
Comment (madamczykhabana) [inline vllm/worker/habana_model_runner.py:250]: We can drop hltv support. There's a debug flag available that will the profiler return both .json.gz and .hltv at the same time

File: vllm/worker/habana_model_runner.py [hunk 3]
```diff
@@ -210,6 +211,56 @@ def align_workers(value, op):
...
+
+
+def setup_profiler():
+    prof_wait = 0
+    prof_warmup = 2
+    prof_active = 1
+    prof_type = os.environ.get('VLLM_PT_PROFILE_METHOD', 'pt')
```
Comment (madamczykhabana) [inline vllm/worker/habana_model_runner.py:257]: as above

File: vllm/worker/habana_model_runner.py [hunk 4]
```diff
@@ -1314,10 +1365,20 @@ def warmup_scenario(self,
                 for i, b in enumerate(blocks)
             ]
         torch.hpu.synchronize()
+        profiler = None
+        if is_profile_run and self.is_driver_worker:
+            profiler = setup_profiler()
+            profiler.start()
+        self.profiler.start('internal', scenario_name)
```
Comment (madamczykhabana) [inline vllm/worker/habana_model_runner.py:1372]: It's already called on line 1336

**Review by madamczykhabana — CHANGES_REQUESTED** [review]

File: vllm/worker/habana_model_runner.py (hunk 2 above)
Comment (adobrzyniewicz-habana) [reply vllm/worker/habana_model_runner.py]: Dropped

File: vllm/worker/habana_model_runner.py (hunk 3 above)
Comment (adobrzyniewicz-habana) [reply vllm/worker/habana_model_runner.py]: Removed, moved to one mmethod

File: vllm/worker/habana_model_runner.py (hunk 4 above)
Comment (adobrzyniewicz-habana) [reply vllm/worker/habana_model_runner.py]: Removed

**Review by madamczykhabana — APPROVED** [review]
LGTM

**Pull request merged**